source/mir/random/engine/mersenne_twister.d

```diff
@@ -58,11 +58,12 @@ struct MersenneTwisterEngine(UIntType, size_t w, size_t n, size_t m, size_t r,
     /// The default seed value.
     enum UIntType defaultSeed = 5489;
 
-    private UIntType _z;
     /++
     Current reversed payload index with initial value equals to `n-1`
     +/
-    private UIntType index = void;
+    ptrdiff_t index = void;
```

Comment:
> It is public to allow user to track Mt state. User may want to initialise it.

That seems an odd use-case, to be honest.  As far as initialization is concerned it might be better to provide a means for the user to seed or initialize the generator based on explicit state variables (with clear "use at your own risk" documentation) rather than breaking encapsulation in this way.  (It's not obvious to me that, in your implementation, it could ever be correct to set the index value alone without also ensuring there is a valid `data` array in place.)

As far as _getting_ is concerned, it might be preferable to provide a property method rather than allowing public access to the underlying variable.  Also, note that this implementation's copy of "index" is exactly the opposite of what the index would be in most other implementations of Mersenne Twister, so it might lead to some misleading comparisons if the user is not strongly aware of the implementation details.